integration/spark/shared/src/main/java/io/openlineage/spark/api/SparkOpenLineageConfig.java

```diff
@@ -26,51 +26,53 @@
 public class SparkOpenLineageConfig extends OpenLineageConfig<SparkOpenLineageConfig> {
 
   public static final String DEFAULT_NAMESPACE = "default";
-  public static final String[] DEFAULT_DISABLED_FACETS =
-      new String[] {"spark_unknown", "spark.logicalPlan"};
-  public static final String DEFAULT_DEBUG_FACET = "disabled";
+  public static final String[] DEFAULT_DISABLED_FACETS = new String[] {};
 
   @Setter @NonNull private String namespace;
   @Setter @Getter private String parentJobName;
   @Setter @Getter private String parentJobNamespace;
   @Setter @Getter private String parentRunId;
   @Setter @Getter private String overriddenAppName;
-  @Setter @NonNull private String debugFacet;
   @Setter @Getter private String testExtensionProvider;
   @Setter private JobNameConfig jobName;
   @Setter private JobConfig job;
 
+  @JsonProperty("facets")
+  @Setter
+  protected SparkFacetsConfig facetsConfig;
+
   public SparkOpenLineageConfig(
       String namespace,
       String parentJobName,
       String parentJobNamespace,
       String parentRunId,
       String overriddenAppName,
-      String debugFacet,
       String testExtensionProvider,
       JobNameConfig jobName,
       JobConfig job,
       TransportConfig transportConfig,
-      FacetsConfig facetsConfig,
+      SparkFacetsConfig facets,
       DatasetConfig datasetConfig,
       CircuitBreakerConfig circuitBreaker,
       Map<String, Object> metricsConfig) {
-    super(transportConfig, facetsConfig, datasetConfig, circuitBreaker, metricsConfig);
+    super(transportConfig, facets, datasetConfig, circuitBreaker, metricsConfig);
     this.namespace = namespace;
     this.parentJobName = parentJobName;
     this.parentJobNamespace = parentJobNamespace;
     this.parentRunId = parentRunId;
     this.overriddenAppName = overriddenAppName;
-    this.debugFacet = debugFacet;
     this.testExtensionProvider = testExtensionProvider;
     this.jobName = jobName;
     this.job = job;
+    this.facetsConfig = facets;
   }
 
   @Override
-  public FacetsConfig getFacetsConfig() {
+  public SparkFacetsConfig getFacetsConfig() {
     if (facetsConfig == null) {
-      facetsConfig = new FacetsConfig();
+      // the configuration creates the default values in the getters, when they are accessed for the
```

Comment:
I spent some time trying to understand where the facets config came from when I deserialised the configuration without these properties and had nulls there, but when accesses I saw values. I continue the same pattern for the facets configuration class.
We are ok with the nulls there. When someone accesses them, the values are going to be created with some default values.